src/runtime/numerics.rs

```diff
@@ -1,1053 +1,819 @@
 use windows::foundation::numerics::{Matrix3x2, Matrix4x4, Vector2, Vector3, Vector4};
 
 // Vector2
-impl ::std::ops::Add<Vector2> for Vector2 {
-    type Output = Vector2;
-    fn add(self, rhs: Vector2) -> Vector2 {
-        Vector2 {
+impl Vector2 {
+    pub fn zero() -> Self {
+        Self { x: 0f32, y: 0f32 }
+    }
+    pub fn one() -> Self {
+        Self { x: 1f32, y: 1f32 }
+    }
+    pub fn unit_x() -> Self {
+        Self { x: 1.0, y: 0.0 }
+    }
+    pub fn unit_y() -> Self {
+        Self { x: 0.0, y: 1.0 }
+    }
+    pub fn dot(&self, rhs: &Self) -> f32 {
+        self.x * rhs.x + self.y * rhs.y
+    }
+    pub fn length_squared(&self) -> f32 {
+        self.dot(self)
+    }
+    pub fn length(&self) -> f32 {
+        self.length_squared().sqrt()
+    }
+    pub fn distance(&self, value: &Self) -> f32 {
+        (self - value).length()
+    }
+    pub fn distance_squared(&self, value: &Self) -> f32 {
+        (self - value).length_squared()
+    }
+    pub fn normalize(&self) -> Self {
+        self / self.length()
+    }
+
+    fn impl_add(&self, rhs: &Self) -> Self {
```

Comment:
What's the purpose of these `impl_` methods?

Reply:
They're the common implementations of the various `add` methods in the traits that follow. It can't simply be called `add` because that confuses the compiler into thinking it might be recursive.